name: Qiita Sync Check

on:
    schedule:
        - cron: "29 17 * * *"
    workflow_run:
        workflows: ["Qiita Sync"]
        types:
            - completed
    workflow_dispatch:

jobs:
    qiita_sync_check:
        name: qiita-sync check
        runs-on: ubuntu-latest
        steps:
            - name: Checkout
              uses: actions/checkout@v2

            - name: Set up Python
              uses: actions/setup-python@v2
              with:
                  python-version: "3.9"

            - name: Install qiita-sync
              run: |
                  python -m pip install qiita-sync

            - name: Run qiita-sync check
              run: |
                  qiita_sync check . > ./qiita_sync_output.txt || true
                  cat ./qiita_sync_output.txt
                  if [ -s "./qiita_sync_output.txt" ]; then
                    echo "Errors detected in qiita_sync check."
                    exit 1
                  fi
              env:
                  QIITA_ACCESS_TOKEN: ${{ secrets.QIITA_ACCESS_TOKEN }}

            - name: Save qiita_sync_output.txt as artifact
              if: failure()
              uses: actions/upload-artifact@v3
              with:
                  name: qiita-sync-output
                  path: ./qiita_sync_output.txt
